frida_tools/repl.py

```diff
@@ -78,6 +81,7 @@ def _add_options(self, parser):
             parser.add_argument("-o", "--output", help="output to log file", dest="logfile")
             parser.add_argument("--eternalize", help="eternalize the script before exit", action='store_true', dest="eternalize", default=False)
             parser.add_argument("--exit-on-error", help="exit with code 1 after encountering any exception in the SCRIPT", action='store_true', dest="exit_on_error", default=False)
+            parser.add_argument("--auto-perform", help="wrap entered code with Java.perform", action='store_true', dest="autoperform", default=False)
```

Comment:
This is very specific to one runtime, while the name doesn't suggest it. I'm wondering if we should make this more generic, so it also does the right thing when in e.g. an iOS app, where `ObjC.schedule(ObjC.mainQueue, ...)` could be used. I'm a bit reluctant to add an option for this though, and tempted to just do it automatically, with a way to opt out. (Like a quick way to switch the current "context", and have it indicated by the prompt perhaps?)

Reply:
Notice this is a code from a different PR #76, I just based this PR on the other one to reduce conflicts.

Regarding the comment, I'm afraid that because the auto perform wrapping isn't perfect it should not be enabled by default, at least not with the current implementation. Variables declared with `var foo` or `let bar` for examples won't survive between different `Java.perform` calls. But we can make it wrap ObjC as well but I don't have an iOS/MacOS device available so won't be able to do it myself :(